public/app/features/dashboard/api/UnifiedDashboardAPI.ts

```diff
@@ -47,4 +49,44 @@ export class UnifiedDashboardAPI
   async deleteDashboard(uid: string, showSuccessAlert: boolean) {
     return await this.v1Client.deleteDashboard(uid, showSuccessAlert);
   }
+
+  /**
+   * List deleted dashboards handling mixed v1/v2 versions or pure v2 dashboards.
+   *
+   * Steps:
+   * 1. Call v1 client to get all deleted dashboards
+   * 2. Check if any items have failed conversion from v2 versions
+   * 3. If v2 dashboards detected (mixed or pure v2), also call v2 client
+   * 4. Filter and combine responses to avoid conversion errors
+   */
+  async listDeletedDashboards(
```

Comment:
@axelavargas after speaking to app-platform I made some changes to handle the mixed responses since both v1 and v2 endpoints return conversion error for the items with a different version (details in the comment).